### PR TITLE
6972 my info speaker

### PR DESCRIPTION
**Pull request by AlexKutkanych**

### CODE REVIEW

File: conference/react-js/src/components/MyInfo/MyInfo.jsx [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+      showPreventUnsavedExitModal: false,
+      showInfoSavedModal: false,
+      showChangePhotoModal: false,
+      user: {},
+    };
+  }
+
+  componentDidMount() {
+    this.setDefaultValues(this.props);
+    console.log(this.props.user);
```
Comment (SupraVlad) [inline conference/react-js/src/components/MyInfo/MyInfo.jsx]: remove, please.

File: conference/react-js/src/components/MyInfo/MyInfo.jsx [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+    e.preventDefault();
+
+    const saveUserInfo = ({ bio, job, company }) => {
+      const body = { bio, job, company };
+      return axios.post(myInfo, body);
+    };
+
+    saveUserInfo(this.state.user)
+      .then((res) => {
+        console.log(res);
```
Comment (SupraVlad) [inline conference/react-js/src/components/MyInfo/MyInfo.jsx]: same. remove logging from final version

File: conference/react-js/src/components/MyInfo/MyInfo.jsx [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+            labelClassName="my-info__label"
+            className="textarea textarea-focus"
+            label="Past Conferences"
+            name="past"
+            rows={5}
+            maxLen={1000}
+            value={past}
+            onChange={this.handleInput}
+          />
+          {/* <InputBlock */}
```
Comment (SupraVlad) [inline conference/react-js/src/components/MyInfo/MyInfo.jsx]: remove comments for final version

File: conference/react-js/src/components/MyInfo/my-info.scss [hunk 4]
```diff
@@ -0,0 +1,115 @@
+.my-info {
+  font-family: "Oswald Light", sans-serif;
+  position: relative;
+
+  &__ava {
```
Comment (SupraVlad) [inline conference/react-js/src/components/MyInfo/my-info.scss]: it will be better to avoid unclear abbreviations

File: .gitignore [hunk 5]
```diff
@@ -196,4 +196,4 @@ hs_err_pid*
 coverage
 
 #file for start localhost#
-tomcat.sh
\ No newline at end of file
+tomcat.sh
```
Comment (SupraVlad) [inline .gitignore:199]: seem like `tomcat.sh` should be excluded but it is still on the changelist. should it be here or not?

**Review by SupraVlad — CHANGES_REQUESTED** [review]

File: conference/react-js/src/components/MyInfo/MyInfo.jsx (hunk 2 above)
Comment (AlexKutkanych) [reply conference/react-js/src/components/MyInfo/MyInfo.jsx]: backend part is not ready for this post request yet, so I need something to know if there is an error or not

File: conference/react-js/src/components/MyInfo/MyInfo.jsx [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+    e.preventDefault();
+    this.setState({ showInfoSavedModal: false });
+  };
+
+  handleCloseModal1 = (e) => {
+    e.preventDefault();
+    this.setState({ showChangePhotoModal: false });
+  };
+
+  handleInput = (e) => {
```
Comment (ScreamCat1) [inline conference/react-js/src/components/MyInfo/MyInfo.jsx]: I think, you should rewrite this function with destructor, and you don`t need use e.preventDefault()
Such as: "handleInput({target: {name, value })=> { this.setState(prevState => ({...prevState.user, name: value}))"

File: conference/react-js/src/components/MyInfo/MyInfo.jsx [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+      showChangePhotoModal: false,
+      user: {},
+    };
+  }
+
+  componentDidMount() {
+    this.setDefaultValues(this.props);
+  }
+
+  setDefaultValues = ({ user }) => {
```
Comment (ScreamCat1) [inline conference/react-js/src/components/MyInfo/MyInfo.jsx]: I think, u should rename this function

File: conference/react-js/src/components/MyInfo/MyInfo.jsx [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+    };
+
+    // update myInfo via post request
+    saveUserInfo(this.state.user);
+
+    this.setState({ showInfoSavedModal: true });
+  };
+
+  handleChangePhoto = (e) => {
+    e.preventDefault();
```
Comment (ScreamCat1) [inline conference/react-js/src/components/MyInfo/MyInfo.jsx]: Same preventDefault

File: conference/react-js/src/components/MyInfo/PopUps/1 [hunk 9]
```diff
@@ -0,0 +1,25 @@
+<h3 className="pop-up__title">Upload New Photo</h3>
```
Comment (ScreamCat1) [inline conference/react-js/src/components/MyInfo/PopUps/1:1]: Delete this file plz

File: conference/react-js/src/components/MyInfo/PopUps/PopUpPreventUnsavedExit.jsx [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+
+class PopUpPreventUnsavedExit extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showModal: false,
+    };
+  }
+
+  handleOpenModal = () => {
```
Comment (ScreamCat1) [inline conference/react-js/src/components/MyInfo/PopUps/PopUpPreventUnsavedExit.jsx]: I think u should rewrite this functions into one function such as toggle

File: conference/react-js/src/reducers/contacts.js [hunk 11]
```diff
@@ -0,0 +1,12 @@
+import actionTypes from '../constants/actions-types';
+
+const contacts = (state = [], { type, payload }) => {
+  const { SET_CONTACTS } = actionTypes;
```
Comment (ScreamCat1) [inline conference/react-js/src/reducers/contacts.js]: I think u should replace before contacts

**Review by ScreamCat1 — APPROVED** [review]

**Review by SupraVlad — APPROVED** [review]

**Pull request merged**